internal/app/machined/pkg/controllers/siderolink/manager.go

```diff
@@ -0,0 +1,198 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package siderolink
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/AlekSi/pointer"
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/talos-systems/go-procfs/procfs"
+	"github.com/talos-systems/go-smbios/smbios"
+	pb "github.com/talos-systems/siderolink/api/siderolink"
+	"go.uber.org/zap"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+	"google.golang.org/grpc"
+	"inet.af/netaddr"
+
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+	"github.com/talos-systems/talos/pkg/machinery/nethelpers"
+	"github.com/talos-systems/talos/pkg/machinery/resources/network"
+)
+
+// ManagerController interacts with SideroLink API and brings up the SideroLink Wireguard interface.
+type ManagerController struct {
+	Cmdline *procfs.Cmdline
```

Comment:
minor nit: probably it should use the same letter case as we already have in  `TimeServerConfigController`: `Cmdline`.